[3.x]Update THIRD-PARTY-LICENSES.txt

himani2411 opened the pull request:
### Description of changes
Making changes with respect to Opens source Distribution for adding specified URL's and License files.


### Tests
* No tests. Running Locally



### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Update THIRD-PARTY-LICENSES.txt

Making changes with respect to Opens source Distribution for adding specified URL's and License files

Commit:
Update create-attribution-doc.sh and THIRD-PARTY-LICENSES.txt

Making changes with respect to Opens source Distribution for adding specified URL's and License files

dreambeyondorange commented on util/create-attribution-doc.sh:
```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
-# Appending Python License to THIRD-PARTY-LICENSES file
-curl https://raw.githubusercontent.com/python/cpython/$cpy_version/LICENSE >> $final_license_file
+# Function to Append Licenses of different Packages to the THIRD-PARTY-LICENSES file
+append_to_final_license_file(){
```
so is this function not actually adding the content of the license file, just the URL?  Don't we need to include the full license text?

dreambeyondorange replied:
I think we should be specific about this in the name about what exactly we are appending

himani2411 replied:
Its appending a header which is Package Name, Package Version, License Type , URL for package to the final License file as well as the whole License file. I have included comments inside the function

dreambeyondorange commented on util/create-attribution-doc.sh:
```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
+# Appending pyinstaller-hooks-contrib 3 License files 
+wget -c https://github.com/pyinstaller/pyinstaller-hooks-contrib/archive/refs/tags/2022.15.tar.gz -O - | tar -xz
+
+cat pyinstaller-hooks-contrib-2022.15/LICENSE >> $final_license_file
```
why only do this for the hooks package and not the others we are ignoring with pip-licenses?

himani2411 replied:
The distribution team provided this specific link for the tar file, which is why I need to unzip it and add to the final file.  For others its a direct link which I can use.

himani2411 replied:
I could directly use this link https://github.com/pyinstaller/pyinstaller-hooks-contrib/tree/2022.15, But I thought we had already decided on going with the approach of using the tar file and unzipping it.

himani2411 replied:
I have changed the link and directly taking from online using  https://github.com/pyinstaller/pyinstaller-hooks-contrib/tree/2022.15, so that I can reuse the same function

dreambeyondorange replied:
What I mean is it seems like we are just putting a link for the license instead of the actual license content for these packages, whereas with the hooks package we put the license content into the file.  Is that correct?

himani2411 replied:
No we are putting the license content for all the packages into the final file as well as the License link

dreambeyondorange commented on util/create-attribution-doc.sh:
```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
+
+# pyinstaller
+# Adding a header for pyinstaller version and License URL 
+echo -e "\n\n\npyinstaller \n5.7.0\nGNU General Public License v2 (GPLv2) \nhttps://github.com/pyinstaller/pyinstaller/archive/refs/tags/v5.7.0.tar.gz." >> $final_license_file
```
should we use the function for this?

himani2411 replied:
Sure

dreambeyondorange commented on util/create-attribution-doc.sh:
```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
+echo -e "\n\n\npyinstaller \n5.7.0\nGNU General Public License v2 (GPLv2) \nhttps://github.com/pyinstaller/pyinstaller/archive/refs/tags/v5.7.0.tar.gz." >> $final_license_file
+wget -c https://github.com/pyinstaller/pyinstaller/archive/refs/tags/v5.7.0.tar.gz -O - | tar -xz
+# Appending pyinstaller License file
+cat pyinstaller-5.7.0/COPYING.txt >> $final_license_file
```
is this file not available in the other archive above?

himani2411 replied:
No I couldnt find this ( COPYING.txt) for other packages.

Commit:
Update create-attribution-doc.sh and THIRD-PARTY-LICENSES.txt

Making changes with respect to Opens source Distribution for adding specified URL's and License files

himani2411 committed:
Merge branch 'develop' into wip/attribution-update
